inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php

```diff
@@ -612,13 +623,20 @@ public function check_job_status( int $id ) {
 			if ( $row_details->retries >= 3 ) {
 				Logger::debug( 'RUCSS: Job failed 3 times for url: ' . $row_details->url );
 
-				$this->used_css_query->make_status_failed( $id );
+				$this->used_css_query->make_status_failed( $id, $job_details['message'] );
 
 				return;
 			}
 
-			// Increment the retries number with 1 and Change status to pending again.
-			$this->used_css_query->increment_retries( $id, $row_details->retries );
+			// on timeout errors with code 504 create new job.
+			if ( 504 === $job_details['code'] ) {
+				$add_to_queue_response = $this->add_url_to_the_queue( $row_details->url, $row_details->is_mobile );
+				if ( false !== $add_to_queue_response ) {
+					$new_job_id = $add_to_queue_response['contents']['jobId'];
+				}
+			}
+			// Increment the retries number with 1 , Change status to pending again and change job id on timeout.
+			$this->used_css_query->increment_retries( $id, $row_details->retries, $new_job_id );
```

Comment:
We need to apply single responsibility concept here, so the method `increment_retries` is used to increment the number of retries, I know that you need to only make one query to handle both cases, but think of it, the normal case, this new_job_id would be false.

Also, timeout status code is 408.

What do u think?

```suggestion
			// on timeout errors with code 408 create new job.
			switch( $job_details['code'] ){
				case 408:
					$add_to_queue_response = $this->add_url_to_the_queue( $row_details->url, $row_details->is_mobile );
					if ( false !== $add_to_queue_response ) {
						// Create a new method in the query class to update the job_id column with $add_to_queue_response['contents']['jobId']
					}
				break;
			}
			// Increment the retries number with 1 , Change status to pending again and change job id on timeout.
			$this->used_css_query->increment_retries( $id, $row_details->retries );
```